service-api/module/Application/src/Controller/YotiController.php

```diff
@@ -56,22 +63,30 @@ public function findPostOfficeAction(): JsonModel
         $this->getResponse()->setStatusCode(Response::STATUS_CODE_200);
         return new JsonModel($branches);
     }
-    public function createSessionAction(array $sessionData): JsonModel
+    public function createSessionAction(): JsonModel
     {
         $uuid = $this->params()->fromRoute('uuid');
-        //@TODO authenticate if not using mock?
-        $result = $this->yotiService->createSession($sessionData);
-        //save sessionId back to caseData
-        if ($result["status"] < 400) {
-            $this->dataImportHandler->updateCaseData(
-                $uuid,
-                'sessionId',
-                'S',
-                $result["data"]["session_id"]
-            );
+
+        if (! $uuid) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_400);
+            return new JsonModel(['error' => 'Missing uuid']);
+        }
+        $authToken = strval(Uuid::uuid4());
+        $caseData = $this->dataQuery->getCaseByUUID($uuid);
+        $sessionData = $this->sessionConfig->build($caseData, $authToken);
+
+        try {
+            $result = $this->yotiService->createSession($sessionData);
+        } catch (YotiException $e) {
+            $this->getResponse()->setStatusCode(Response::STATUS_CODE_500);
+            return new JsonModel(new Problem(
+                'Problem with request',
```

Comment:
Probably better to be more explicit here. If something goes wrong, we're likely to end up with this text as the "culprit" so it's helpful if we can immediately identify that it's a Yoti problem (speaking from experience of trying to debug a problem in Sirius where we just knew "an unexpected error occurred" 😄)
```suggestion
                'Problem requesting Yoti API',
```